src/theory/builtin/theory_builtin_rewriter.cpp

```diff
@@ -302,11 +308,12 @@ Node TheoryBuiltinRewriter::getArrayRepresentationForLambdaRec( TNode n, bool re
   }
 }
 
-Node TheoryBuiltinRewriter::getArrayRepresentationForLambda( TNode n, bool reqConst ){
+Node TheoryBuiltinRewriter::getArrayRepresentationForLambda(TNode n)
+{
   Assert( n.getKind()==kind::LAMBDA );
   // must carry the overall return type to deal with cases like (lambda ((x Int)(y Int)) (ite (= x _) 0.5 0.0)),
   //  where the inner construction for the else case about should be (arraystoreall (Array Int Real) 0.0)
-  return getArrayRepresentationForLambdaRec( n, reqConst, n[1].getType() );
```

Comment:
Can we add an assertion that the return value of `getArrayRepresentationForLambdaRec` is either null or constant?

Reply:
Actually my documentation wasn't accurate. I've updated it.  The method ensures that the array is almost constant, but isn't necessary a canonical array constant.